src/System.Diagnostics.Process/tests/ProcessCollectionTests.cs

```diff
@@ -12,7 +12,7 @@ public class ProcessCollectionTests : ProcessTestBase
         [Fact]
         public void TestModuleCollectionBehavior()
         {
-            ProcessModule[] mArray = _process.Modules.Cast<ProcessModule>().ToArray();
+            ProcessModule[] mArray = Process.GetCurrentProcess().Modules.Cast<ProcessModule>().ToArray();
```

Comment:
Hmm, taking a closer look at the line where this is thrown, it could be because even after 50 tries, the process modules weren't loaded or the OS loader is touching the module information during this time. For this case too, win32 api throws an exception, with the same error message that is ```Only part of a ReadProcessMemory or WriteProcessMemory request was completed``` as the bitness problem.

Either way, using the CurrentProcess Modules should solve both these problems. One scenario where the CurrentProcess may fail will be when the OSLoader is touching the module information of the current process. But in the 50 iterations we should have a success hit, as another test with similar failure before has stopped failing with this change.